src/commands/new.ts

```diff
@@ -0,0 +1,133 @@
+// TODO: Can we avoid disabling all of these rules?
+/* eslint-disable @typescript-eslint/no-unsafe-member-access -- Prompts breaks this but its an external module so there's nothing we can do about it  */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment -- Prompts breaks this but its an external module so there's nothing we can do about it */
+
+import { basename, dirname } from "path";
+import { Command, flags } from "@oclif/command";
+import prompts from "prompts";
+import { getStackNameFromFileName } from "../utils/args";
+import type { CDKTemplate } from "../utils/cdk";
+import { construct } from "../utils/cdk";
+import { Imports } from "../utils/imports";
+
+interface NewCommandConfig {
+  outputPath: string;
+  outputDir: string;
+  outputFile: string;
+  stackName: string;
+}
+
+interface NewCommandArgs {
+  output: string;
+  stack?: string;
+}
+
+export class NewCommand extends Command {
+  static description = "Creates a new CDK stack";
+
+  static flags = {
+    version: flags.version({ char: "v" }),
+    help: flags.help({ char: "h" }),
+  };
+
+  static args = [
+    {
+      name: "output",
+      required: true,
+      description: "The file to output CDK to",
+    },
+    {
+      name: "stack",
+      required: false,
+      description: "A name to give the stack. Defaults to match the filename.",
+    },
+  ];
+
+  imports = new Imports();
+
+  template: CDKTemplate = {
+    Parameters: {},
+  };
+
+  static getConfig = ({ args }: { args: NewCommandArgs }): NewCommandConfig => {
+    const outputFile = basename(args.output);
+
+    return {
+      outputPath: args.output,
+      outputDir: dirname(args.output),
+      outputFile: outputFile,
```

Comment:
Testing this out locally, we are currently able to create files with no file extension specified (eg I created file `foo` and the file was called `foo`, not `foo.ts`. Since we will always be writing these files in Typescript (at least for now) could we do this instead? Tested locally and seems to be working.


```suggestion
      outputFile: outputFile.endsWith("ts") ? outputFile : `${outputFile}.ts`,
;
```

Reply:
Good idea! I've implemented this plus a test for it